app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,598 @@
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.net.Uri;
+import android.net.wifi.p2p.WifiP2pDevice;
+import android.net.wifi.p2p.WifiP2pDeviceList;
+import android.net.wifi.p2p.WifiP2pInfo;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnItemClick;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.util.List;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function0;
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.KiwixApplication;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.AlertDialogShower;
+import org.kiwix.kiwixmobile.utils.KiwixDialog;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.TO_BE_SENT;
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019.
+ *
+ * This activity is the starting point for the module used for sharing zims between devices.
+ *
+ * The module is used for transferring ZIM files from one device to another, from within the
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using {@link PeerGroupHandshakeAsyncTask}
+ * 2) After handshake, starting the files transfer using {@link SenderDeviceAsyncTask} on the sender
+ * device and {@link ReceiverDeviceAsyncTask} files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning")
+public class LocalFileTransferActivity extends AppCompatActivity implements
+    WifiDirectManager.Callbacks {
+
+  // Not a typo, 'Log' tags have a length upper limit of 25 characters
+  public static final String TAG = "LocalFileTransferActvty";
+  public static final int REQUEST_ENABLE_WIFI_P2P = 1;
+  public static final int REQUEST_ENABLE_LOCATION_SERVICES = 2;
+  private static final int PERMISSION_REQUEST_CODE_COARSE_LOCATION = 1;
+  private static final int PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS = 2;
+
+  public static int PEER_HANDSHAKE_PORT = 8009;
```

Comment:
These constants should almost certainly not be in the activity

Reply:
Should I place them to the Constants file?
Also, is this because all constants should be centrally placed irrespective of whether their use is restricted to some particular set of files?

Reply:
A Constants file is generally the last resort, a constant should live in the class where it belongs and that helps its discoverability the most. Take for example the IntentFilter Actions in WifiDirectManager, these actions live in WifiP2PManager because they are most related to it. I think WifiDirectManager is the best shout.

Reply:
- Peer handshake port was only needed in its async task, shifted it there
- File transfer was required in both sender and receiver async tasks, so shifted it to WDM
- Rest are required in LFTA and others, so left them as is
[Commit](https://github.com/kiwix/kiwix-android/pull/1276/commits/d02c940baa790428524ce3940303e39487e76eed)